### PR TITLE
24 poc of implementation of discord service

### LINKED ISSUE #24: POC of implementation of Discord service
**Short description:**
Add the service Discord using the actual architecture.

**Describe the solution you'd like**
Connect to a Discord account, get some events and send messages.

### PR DESCRIPTION
**Pull request by tbellicha**
# Purpose

The purpose of this pull request is to add the service Discord.

Fixes #24 

## Added features

* This service has been added to the application. It includes:
  * A reaction which send a private message
  * A reaction which send a message in a channel
  * A reaction which change activity
  * An action which is activated on joining a voice channel
  * An action which is activated on reacting to a message
  * An action which is activated on sending a message in a channel

## Updated features

* Usage of Discord and Discord.js dependencies in the package.json

### CODE REVIEW

File: server/services/discord/actions/on_join_voice_channel.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+ * @param {*} oldChannel Previous channel object
+ * @param {*} newChannel Current channel object
+ */
+client.on('voiceStateUpdate', (oldChannel, newChannel) => {
+  var oldUserChannel = oldChannel.channelID
+  var newUserChannel = newChannel.channelID
+  if (!newUserChannel) console.log('User left the channel') //  Leaving a voice channel
+  if (newUserChannel === 'ID of the channel wanted')
+    //  If it's the channel that you want
+    console.log('Joined the channel wanted')
```
Comment (LucasTesnier) [inline server/services/discord/actions/on_join_voice_channel.js:21]: We need to implement trigger to replace .log

File: server/services/discord/actions/on_reaction_add.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+'use strict'
+
+const client = require('../connect')
+
+/**
+ * @brief Triggered when a user reacts to a message.
+ * @param {*} reaction Reaction object
+ * @param {*} user User that reacted to the message
+ */
+client.on('messageReactionAdd', (reaction, user) => {
```
Comment (LucasTesnier) [inline server/services/discord/actions/on_reaction_add.js:10]: Trigered in every discord server and MP ?

**Review by LucasTesnier — CHANGES_REQUESTED** [review]
Answer to my question and add the service, action, and reaction to the Database

File: server/services/discord/actions/on_reaction_add.js (hunk 2 above)
Comment (tbellicha) [reply server/services/discord/actions/on_reaction_add.js:10]: Yes that's it. So when we will use it, we will need to check the server and channels/messages IDs if needed

**Review by LucasTesnier — APPROVED** [review]

**Review by TomDUVAL-MAHE — APPROVED** [review]
NTR

**Review by ZQUEMA — APPROVED** [review]

**Pull request merged**